frontend/public/components/routes.tsx

```diff
@@ -200,26 +203,49 @@ const RouteListRow: React.SFC<RoutesRowProps> = ({obj: route}) => <ResourceRow o
   <div className="col-lg-2 hidden-md hidden-sm hidden-xs"><RouteStatus obj={route} /></div>
 </ResourceRow>;
 
-const TLSSettings = props => <span>
-  {!props.tls && 'TLS is not enabled.'}
-  {props.tls && <dl>
-    <dt>Termination Type</dt>
-    <dd>{props.tls.termination}</dd>
-    <dt>Insecure Traffic</dt>
-    <dd>{props.tls.insecureEdgeTerminationPolicy || '-'}</dd>
-    <dt>Certificate</dt>
-    <dd>{props.tls.certificate || '-'}</dd>
-    <dt>Key</dt>
-    <dd>{props.tls.key || '-'}</dd>
-    <dt>CA Certificate</dt>
-    <dd>{props.tls.caCertificate || '-'}</dd>
-    {_.get(props.tls, 'termination') === 'reencrypt' && <span>
-      <dt>Destination CA Cert</dt>
-      <dd>{props.tls.destinationCACertificate || '-'}</dd>
-    </span>}
-  </dl>
+class TLSSettings extends SafetyFirst<TLSDataProps, TLSDataState> {
+  constructor(props) {
+    super(props);
+    this.state = { showPrivateKey: false };
+    this.toggleKey = this.toggleKey.bind(this);
+  }
+
+  toggleKey() {
+    this.setState({ showPrivateKey: !this.state.showPrivateKey });
+  }
+
+  render() {
+    const { showPrivateKey } = this.state;
+    const { tls } = this.props;
+    const visibleKeyValue = showPrivateKey ? tls.key : <MaskedData /> ;
+
+    return !tls ? 'TLS is not enabled.'
+      : <dl>
```

Comment:
Care to indent the `<dt>` under the `<dl>`?

```JavaScript 
return !tls ?
  'TLS is not enabled' :
  <dl>
      <dt>stuff</dt>
      <dt>things</dt>
  </dl>;

```

Reply:
fixed